clients/imodels-client-management/src/base/internal/AxiosRestClient.ts

```diff
@@ -82,3 +98,34 @@ export class AxiosRestClient implements RestClient {
     }
   }
 }
+
+/** Default implementation for {@link HttpResponseHeadersFactory}. */
+export class AxiosHeadersAdapterFactory implements HttpResponseHeadersFactory {
+  public create(response: AxiosResponse): HttpResponseHeaders {
+    return new AxiosHeadersAdapter(response);
+  }
+}
+
+/** Default implementation for {@link HttpResponseHeaders} interface, which adapts `axios` HTTP response headers to headers expected by the iModels Client. */
+export class AxiosHeadersAdapter implements HttpResponseHeaders {
+  private _response: AxiosResponse;
+
+  constructor(response: AxiosResponse) {
+    this._response = response;
+  }
+
+  public get(headerName: string): unknown {
+    // Directly manipulating headers object is deprecated.
+    // https://github.com/axios/axios?tab=readme-ov-file#-axiosheaders
+    if (this._response.headers.get instanceof Function)
```

Comment:
does the type of `this._response.headers.get` depend on specific version on axios used? why can it be both a function and simple object?

Reply:
Yes, it depends on the axios. The method `get` exists on `AxiosHeaders` class, which was introduced in newer versions (https://github.com/axios/axios?tab=readme-ov-file#-axiosheaders).
![image](https://github.com/iTwin/imodels-clients/assets/80688723/837609d5-bc49-4773-9107-960eee5deaac)

Reply:
But we reference a specific version of axios in our packages (~1.6.4), which means that the package manager will install that specific version (or a later patch version). We do not need to support old versions.

Reply:
But that's in axios type definition; we can't call `get` without checking first if it is a function since we get a type error

Reply:
![image](https://github.com/iTwin/imodels-clients/assets/80688723/b81bdcb4-025b-41f8-9706-ca511153f16c)